### DISCUSSION TITLE
Is there a way to get rid of additional properties without throwing?

### DISCUSSION THREAD

**Discussion by errmayank** [sinclairzx81/typebox#588] [Q&A] [answered]
Consider this example

```ts
const data = {
  id: 123,
  name: 'Test',
  email: 'test@test.com',
  timezone: 'Test/Test',
};

const schema = Type.Object(
  {
    id: Type.Number(),
    name: Type.String(),
    email: Type.String(),
    imageUrl: Type.Optional(Type.String()),
  },
  { additionalProperties: false },
);

console.log(Value.Decode(schema, data));
```

This will throw because `timezone` is not defined in the schema, is it possible to decode it without throwing but also omit the extra properties that are not defined in the schema?

**Comment by sinclairzx81** [comment, marked as answer]
@errmayank Hi,

TypeBox requires that the data being decoded (or validated) be correct and it does not automatically remove additional properties during validation or decode phases. This is largely by design as the need to remove additional properties is usually the result of a sender sending invalid data (and should usually be fixed at the sender, not the receiver).

It is possible to write logic to remove any additional properties prior to calling decode, but the extra processing of a value would need to be handled in user code (outside of TypeBox)

Hope this helps
S

**Reply by errmayank** [reply]
Got it, Thanks.
